### PR TITLE
Handle incompatible cluster events errors

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
At some point during workflow execution a scenario may arise when the cluster assigned to an execution is updated. This change introduces failure handling to early abort such workflows still running on the original cluster. IncompatibleCluster errors as they are first received are propagated to the workflow executor which transitions the workflow to failing to clean up resources. This leads to Abort handler calls for running tasks and nodes which swallow event recording failures due to subsequent (and expected) IncompatibleCluster errors.

### Testing
Verified that the workflow CRD metadata ends up with `termination-status: terminated` for IncompatibleCluster errors sent initially for
- Workflow execution events
- Node execution events
- Task execution events 
- Subworkflows
- Child workflows
- Branch nodes

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
See tracking issue for context

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1929

## Follow-up issue
_NA_

### CODE REVIEW

**Review by EngHabu — DISMISSED** [review]
LGTM!

**Review by EngHabu — APPROVED** [review]
👍🏽
